frontend/src/components/filterComponent.js

```diff
@@ -36,6 +38,18 @@ type Props = {
 }
 
 class FilterComponent extends Component<Props> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      workspaces: []
+    }
+  }
+
+  async componentDidMount() {
+    const workspaces = await getWorkspaces()
+    this.setState({ workspaces: workspaces.result })
```

Comment:
will a failed request break `this.state.workspaces.map()`?

Reply:
depends on what you mean by "failed". If you mean that the request returns nothing, then no it doesn't break. If you mean that there's some external error (i.e. network failure), then yes, it will probably break, but this is mostly consistent behavior with other network calls in our web frontend:

https://github.com/hack4impact-uiuc/h4i-recruitment/blob/4878fad7fe1a033f0e48adc42d5cd7e7d2e1a374/frontend/src/pages/stats.js#L27